bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -263,6 +274,12 @@ where
 			BuyExecution { fees: asset_hub_fee_asset, weight_limit: Unlimited },
 			DescendOrigin(PalletInstance(inbound_queue_pallet_index).into()),
 			UniversalOrigin(GlobalConsensus(network)),
+			// No matter what error it is just try to Deposit the asset to beneficiary without
+			// trapped.
+			SetErrorHandler(Xcm(vec![DepositAsset {
+				assets: Definite(asset.clone().into()),
+				beneficiary: beneficiary.clone(),
+			}])),
```

Comment:
Add an extra `SetErrorHandle` try to Deposit the asset to beneficiary in any case.